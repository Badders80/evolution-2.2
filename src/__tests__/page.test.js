import React from "react";
import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import HomePage from "../app/page";

// Mock Next.js Image component
jest.mock("next/image", () => ({
  __esModule: true,
  default: "img",
}));

describe("HomePage", () => {
  it("renders without crashing", () => {
    expect(() => render(<HomePage />)).not.toThrow();
  });

  it("renders the scaffold structure", () => {
    render(<HomePage />);

    // Check main hero heading
    expect(screen.getByRole("heading", { name: "Hero", level: 1 })).toBeInTheDocument();

    // Check section content
    expect(screen.getByText("Section1")).toBeInTheDocument();
    expect(screen.getByText("Section2")).toBeInTheDocument();
    expect(screen.getByText("Section3")).toBeInTheDocument();
    expect(screen.getByText("Section4")).toBeInTheDocument();
    expect(screen.getByText("CTA")).toBeInTheDocument();
    expect(screen.getByText("Footer")).toBeInTheDocument();

    // Check navbar content
    expect(screen.getByText("Nav Bar")).toBeInTheDocument();
    expect(screen.getByText("Get Started / Login")).toBeInTheDocument();

    // Check logo alt text
    expect(screen.getByAltText("Evolution Stables Logo")).toBeInTheDocument();
  });

  it("renders sections with alternating backgrounds", () => {
    render(<HomePage />);

    // Check that all sections are present
    const sections = screen.getAllByText(/^Section\d+$/);
    expect(sections).toHaveLength(4);

    // Check CTA and Footer
    expect(screen.getByText("CTA")).toBeInTheDocument();
    expect(screen.getByText("Footer")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = render(<HomePage />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
