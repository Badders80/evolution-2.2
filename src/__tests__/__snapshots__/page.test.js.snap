// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`HomePage matches snapshot 1`] = `
<div
  class="min-h-screen flex flex-col"
>
  <section
    aria-label="Hero section with Evolution Stables branding"
    class="relative h-screen w-screen flex items-center justify-center"
    role="region"
    style="background-image: url("/assets/Horse-Double-Black.png"); background-size: cover; background-position: center; background-repeat: no-repeat;"
  >
    <div
      class="absolute inset-0 bg-black/50"
    />
    <img
      alt="Evolution Stables branded logo"
      class="relative z-10 w-[60%] max-w-[600px] h-auto drop-shadow-lg"
      height="400"
      src="/assets/ES-Branded-Gold.png"
      width="600"
    />
  </section>
  <div
    class="border-t border-gray-300"
  />
  <section
    aria-label="Section 1 placeholder"
    class="bg-black text-white aspect-[19/6] flex items-center justify-center"
    role="region"
  >
    <h2
      class="text-3xl md:text-5xl font-bw-gradual font-bold"
    >
      Section 1
    </h2>
  </section>
  <div
    class="border-t border-gray-300"
  />
  <section
    aria-label="Section 2 placeholder"
    class="bg-black text-white aspect-[19/6] flex items-center justify-center"
    role="region"
  >
    <h2
      class="text-3xl md:text-5xl font-bw-gradual font-bold"
    >
      Section 2
    </h2>
  </section>
  <div
    class="border-t border-gray-300"
  />
  <section
    aria-label="Call to action section"
    class="bg-black text-white aspect-[19/6] flex items-center justify-center"
    role="region"
  >
    <h2
      class="text-3xl md:text-5xl font-bw-gradual font-bold"
    >
      CTA
    </h2>
  </section>
  <div
    class="border-t border-gray-300"
  />
  <section
    aria-label="Footer section"
    class="bg-black text-white aspect-[19/6] flex items-center justify-center"
    role="region"
  >
    <h2
      class="text-3xl md:text-5xl font-bw-gradual font-bold"
    >
      Footer
    </h2>
  </section>
</div>
`;
